texthook/engine/engine.cc

```diff
@@ -2141,6 +2141,48 @@ bool InsertBGI3Hook()
   return true;
 }
 #endif // 0
+
+bool InsertBGI4Hook()
+{
+    /*
+    * Sample games:
+    * https://vndb.org/v26664
+    */
+    bool found = false;
+    const BYTE pattern[] = {
+        0x55,                           // 55               push ebp
+        0x8b,0xec,                      // 8BEC             mov ebp,esp
+        0x53,                           // 53               push ebx
+        0x56,                           // 56               push esi
+        0x57,                           // 57               push edi
+        0x33, 0xFF,                     // 33 FF            xor edi,edi
+        0xE8, 0x23, 0xFD, 0xFF, 0xFF,   // E8 23FDFFFF      call saclet.exe+A0990
+        0x8B, 0xF0,                     // 8B F0            mov esi,eax
+        0x8B, 0x45, 0x10,               // 8B 45 10         mov eax,[ebp+10]
```

Comment:
The pattern is really long, some of it might change very easily in other games. What about removing everything after 8B F0, does it still hook reasonably well?